main.go

```diff
@@ -115,3 +134,59 @@ func main() {
 		os.Exit(1)
 	}
 }
+
+// getDeploymentNamespace returns the Namespace this operator is deployed on.
+func getDeploymentNamespace() (string, error) {
+	// deployNamespaceEnvVar is the constant for env variable DEPLOYMENT_NAMESPACE
+	// which specifies the Namespace to watch.
+	// An empty value means the operator is running with cluster scope.
+	var deployNamespaceEnvVar = "DEPLOYMENT_NAMESPACE"
+
+	ns, found := os.LookupEnv(deployNamespaceEnvVar)
+	if !found {
+		return "", fmt.Errorf("%s must be set", deployNamespaceEnvVar)
+	}
+	return ns, nil
+}
+
+func createDefaultNHC(c client.Client, namespace string) error {
+	list := remediationv1alpha1.NodeHealthCheckList{}
+	err := c.List(
+		context.Background(),
+		&list,
+		&client.ListOptions{},
+	)
+	if err != nil {
+		return errors.Wrap(err, "failed to list NHC objects")
+	}
+	if len(list.Items) > 0 {
+		// if we have already some NHC then this is a restart, an upgrade, or a redeploy
+		// then we preserve whatever we have.
+		return nil
+	}
+	nhc := remediationv1alpha1.NodeHealthCheck{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: DefaultCRName,
+		},
+		Spec: remediationv1alpha1.NodeHealthCheckSpec{
+			Selector: metav1.LabelSelector{
+				MatchExpressions: []metav1.LabelSelectorRequirement{{
+					Key:      "node-role.kubernetes.io/worker",
+					Operator: metav1.LabelSelectorOpExists,
+				}},
+			},
+			RemediationTemplate: &v1.ObjectReference{
+				Kind:       "PoisonPillRemediationTemplate",
+				APIVersion: "poison-pill.medik8s.io/v1alpha1",
+				Name:       "default-remediation-poisonpill",
```

Comment:
can we change to `default-remediation-template`?

we know it's poison pill by the kind itself